### PR TITLE
Make isochrone 'contour' value serialize with decimals to match input

### PR DESCRIPTION
**Pull request by danpat**
# Issue

Fixes https://github.com/valhalla/valhalla/issues/3094

Adds a new json::float_t type so that we can render decimals in non-fixed format into our API response.
Then, modifies the isochrone serializer to render the `contour` value as a floating point number, which is what we allow on the input side.

As a side-effect, this PR also re-names `json::fp_t` to `json::fixed_t` to help with readability.

## Tasklist

 - [x] Add tests
 - [x] Add #fixes with the issue number that this PR addresses
 - [ ] Update the docs with any new request parameters or changes to behavior described
 - [x] Update the [changelog](CHANGELOG.md)

### CODE REVIEW

File: src/tyr/isochrone_serializer.cc [hunk 1]
```diff
@@ -75,13 +77,13 @@ std::string serializeIsochrones(const Api& request,
...
           {"properties", map({
                              {"metric", std::get<2>(interval)},
-                             {"contour", static_cast<uint64_t>(std::get<1>(interval))},
-                             {"color", hex.str()},            // lines
-                             {"fill", hex.str()},             // geojson.io polys
-                             {"fillColor", hex.str()},        // leaflet polys
-                             {"opacity", fp_t{.33f, 2}},      // lines
-                             {"fill-opacity", fp_t{.33f, 2}}, // geojson.io polys
-                             {"fillOpacity", fp_t{.33f, 2}},  // leaflet polys
+                             {"contour", baldr::json::float_t{std::get<1>(interval)}},
```
Comment (danpat) [inline src/tyr/isochrone_serializer.cc:80]: This is the actual change buried in the noise.

File: valhalla/baldr/json.h [hunk 2]
```diff
@@ -23,20 +23,35 @@ class Jmap;
 using MapPtr = std::shared_ptr<Jmap>;
 class Jarray;
 using ArrayPtr = std::shared_ptr<Jarray>;
-struct fp_t {
+
+// Fixed precision ser
```
Comment (kevinkreiser) [inline valhalla/baldr/json.h:27]: Nit: Did you mean to truncate this?

**Review by kevinkreiser — APPROVED** [review]
I was hoping you'd just ported isochrones serialization to rapidjson, maybe next time

File: valhalla/baldr/json.h (hunk 2 above)
Comment (danpat) [reply valhalla/baldr/json.h:27]: No, I'm sure what happened here is that a kid threw a toy at my head partway through typing or something.  I don't really remember, but I do have a small lump near my temple....

**Comment by danpat** [comment]
> I was hoping you'd just ported isochrones serialization to rapidjson, maybe next time

Duh, I didn't even think of that.  I'll do it as a follow-up PR, there isn't much to untangle on this serializer.

**Review by PureW — APPROVED** [review]

**Pull request merged**